git-tar/Gopkg.toml

```diff
@@ -24,3 +24,6 @@
 [[constraint]]
   name = "github.com/openfaas/faas-cli"
   version = "0.4.20"
+
+[prune]
+  unused-packages = true
```

Comment:
Would be tidier to raise this change in a separate PR so that all the existing vendored deps that get pruned aren't included in this PR.

Reply:
@alexellis is there a project wide view on whether we should prune? It can make development using new packages within a given import a bit messier - bit of an edge case perhaps.

Reply:
Can you explain what you mean? pros/cons of both